src/components/CountrySelector/index.js

```diff
@@ -0,0 +1,155 @@
+import Button from '../Button'
+import { localised } from '../../locales'
+import type { LocalisedType } from '../../locales'
+import { getSupportedCountriesForDocument } from '../../supported-documents'
+import { trackComponent } from 'Tracker'
+import { preventDefaultOnClick } from '~utils/index'
+import theme from 'components/Theme/style.scss'
+import style from './style.scss'
+
+import Autocomplete from 'accessible-autocomplete/preact'
+import 'accessible-autocomplete/dist/accessible-autocomplete.min.css'
+
+type Props = {
+  documentType: string,
+  previousStep: () => void,
+  nextStep: () => void,
+  translate: (string, ?{}) => string,
+  actions: Object,
+} & LocalisedType
+
+type State = {
+  hasError: Boolean,
+}
+
+class CountrySelection extends Component<Props, State> {
+  state = {
+    hasError: false,
+  }
+
+  handleCountrySelect = (selectedCountry: Object) => {
+    if (selectedCountry) {
+      this.setState({
+        hasError: false,
+      })
+      this.props.actions.setIdDocumentIssuingCountry(selectedCountry)
+    } else {
+      this.setState({
+        hasError: true,
+      })
+    }
+  }
+
+  suggestCountries = (query: string = '', populateResults: Function) => {
+    const countries = getSupportedCountriesForDocument(this.props.documentType)
+    const filteredResults = countries.filter((result) => {
+      const country = result.name
+      return country.toLowerCase().includes(query.trim().toLowerCase())
+    })
+    populateResults(filteredResults)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.documentType &&
+      this.props.documentType !== prevProps.documentType
+    ) {
+      this.props.actions.resetIdDocumentIssuingCountry()
+    }
+  }
+
+  render() {
+    const { translate, nextStep, idDocumentIssuingCountry } = this.props
+    return (
+      <div className={theme.fullHeightContainer}>
+        <PageTitle title={translate(`country_selection.title`)} />
+        <div
+          className={classNames(
+            theme.thickWrapper,
+            theme.alignTextLeft,
+            style.container
+          )}
+        >
+          <div data-onfido-qa="countrySelector">
+            <label className={style.label} for="country-finder">
+              {translate(`country_selection.search`)}
+            </label>
+            <Autocomplete
+              id="country-finder"
+              required={true}
+              source={this.suggestCountries}
+              minLength={2}
+              placeholder={translate(`country_selection.placeholder`)}
+              tNoResults={() => translate(`country_selection.error`)}
+              dropdownArrow={() => `<i class="${style.caretIcon}"><i/>`}
+              displayMenu="overlay"
+              templates={{
+                inputValue: (country) => country && country.name,
+                suggestion: (country) =>
+                  country &&
+                  `<span class="${style.countryLabel}">${country.name}</span>`,
+              }}
```

Comment:
This acts as a template string for how the country dropdown options are rendered. Since we're going to add country flags as part of the options display in the next ticket, I just went ahead to get started defining the template here. More details of the library's templating [here](https://github.com/alphagov/accessible-autocomplete).